### PR TITLE
CLI increments to search with both name and description. #1170

### PR DESCRIPTION
**Pull request by isandesh1986**
Adresses/Fixes #<ISSUENO>.

Implements

* 
* 
* 

@devonfw/cobigen
incrementSearchWithName_complex

incrementSearchWithTOS

incrementSearchWithDescription_UC

### CODE REVIEW

File: cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/GenerateCommand.java [hunk 1]
```diff
@@ -561,10 +561,19 @@ private void printFoundArtifacts(ArrayList<GenerableArtifact> possibleArtifacts,
...
 
         for (int i = 0; i < matching.size(); i++) {
-            String description = isIncrements ? ((IncrementTo) matching.get(i)).getDescription()
-                : ((TemplateTo) matching.get(i)).getId();
-            JaccardDistance distance = new JaccardDistance();
-            scores.put(matching.get(i), distance.apply(description.toUpperCase(), userInput.toUpperCase()));
+        	if(!isIncrements) {
+        		String description = ((TemplateTo) matching.get(i)).getId();
+        		JaccardDistance distance = new JaccardDistance();
+                scores.put(matching.get(i), distance.apply(description.toUpperCase(), userInput.toUpperCase()));
```
Comment (maybeec) [inline cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/GenerateCommand.java:567]: just a small formatting issue, not sure why, please check your eclipse editor, that you have run update-all-workspaces to get the correct formattings.

File: cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/GenerateCommand.java [hunk 2]
```diff
@@ -561,10 +561,19 @@ private void printFoundArtifacts(ArrayList<GenerableArtifact> possibleArtifacts,
...
+        	} else {
+        		String description = ((IncrementTo) matching.get(i)).getDescription();
+        		String id = ((IncrementTo) matching.get(i)).getId();
+        		JaccardDistance distance = new JaccardDistance();
+        		Double descriptionDistance = distance.apply(description.toUpperCase(), userInput.toUpperCase());
+        		Double idDistance = distance.apply(id.toUpperCase(), userInput.toUpperCase());
+        		Double finalDistance = (idDistance<descriptionDistance)? idDistance : descriptionDistance;
```
Comment (maybeec) [inline cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/GenerateCommand.java:574]: for readability make use of Math.min()

**Comment by isandesh1986** [comment]
> Please also make sure, that for a perfect match of the id or the descritption (id preferred), we are not taking any other increment / template into account and simply start generating.

Yes, that is already taken care. Have verified it while testing

File: cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/GenerateCommand.java (hunk 1 above)
Comment (isandesh1986) [reply cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/GenerateCommand.java]: Done.

File: cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/GenerateCommand.java (hunk 2 above)
Comment (isandesh1986) [reply cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/GenerateCommand.java]: Done.

**Comment by maybeec** [comment]
@isandesh1986, I think you need to redo the PR.

1. the pull request should be done on the basis and with the target of dev_cli, but it was done on master
2. we it seems that your indentation is strange

**Pull request closed without merging**